### PR TITLE
feat: prevent writes to files in pub cache and FVM folder

### PR DESCRIPTION
**Pull request by sidlatau**

### CODE REVIEW

**Review by akinsho — CHANGES_REQUESTED** [review]
@sidlatau I think all of this could go inside the ftplugin file since it's essentially equivalent to a file type autocommand

**Comment by sidlatau** [comment]
> @sidlatau I think all of this could go inside the ftplugin file since it's essentially equivalent to a file type autocommand

Good point, moved code to ftplugin.

File: ftplugin/dart/init.lua [hunk 1]
```diff
@@ -2,3 +2,11 @@ require("flutter-tools.lsp").attach()
 
 vim.opt_local.comments = [[sO:*\ -,mO:*\ \ ,exO:*/,s1:/*,mb:*,ex:*/,:///,://]]
 vim.opt_local.commentstring = [[//%s]]
+
+local full_path = vim.fn.expand "%:p"
+-- Prevent writes to files in the pub cache and FVM folder.
+if string.find(full_path, ".pub-cache") or
+  string.find(full_path, [[Pub\Cache]]) or
+  string.find(full_path, "/fvm/versions/") then
```
Comment (akinsho) [inline ftplugin/dart/init.lua]:
```suggestion
if full_path:find(".pub-cache") or
  full_path:find([[Pub\Cache]]) or
  full_path:find("/fvm/versions/") then
```

File: ftplugin/dart/init.lua [hunk 2]
```diff
@@ -2,3 +2,11 @@ require("flutter-tools.lsp").attach()
 
 vim.opt_local.comments = [[sO:*\ -,mO:*\ \ ,exO:*/,s1:/*,mb:*,ex:*/,:///,://]]
 vim.opt_local.commentstring = [[//%s]]
+
+local full_path = vim.fn.expand "%:p"
+-- Prevent writes to files in the pub cache and FVM folder.
+if string.find(full_path, ".pub-cache") or
+  string.find(full_path, [[Pub\Cache]]) or
```
Comment (akinsho) [inline ftplugin/dart/init.lua]: I'm not sure if any of these will need to be escaped as special characters in a lua pattern (thinking mostly about the `.` and the `-`). If you've checked them, and they work though, then 👍🏿

Comment (sidlatau) [reply ftplugin/dart/init.lua]: I escaped all the strings, visually looks consistent.

**Review by akinsho — APPROVED** [review]
LGTM 👍🏿

**Pull request merged**